tools/lib/gzipTileset.js

```diff
@@ -40,100 +41,26 @@ function gzipTileset(options) {
 
-    var writeCallback = defaultValue(options.writeCallback, getDefaultWriteCallback(outputDirectory));
+    var writeCallback = defaultValue(options.writeCallback, getDefaultWriteCallback());
     var logCallback = options.logCallback;
 
     if (defined(logCallback)) {
         logCallback((gzip ? 'Compressing' : 'Uncompressing') + ' files...');
     }
 
-    var operation = gzip ? zlib.gzipSync : zlib.gunzipSync;
-
-    return new Promise(function(resolve, reject) {
-        getNumberOfFilesInDirectory(inputDirectory)
-            .then(function(numberOfFiles) {
-                var writeFile = getWriteFile(writeCallback, numberOfFiles, resolve, reject);
-                fsExtra.walk(inputDirectory)
-                    .on('data', function (item) {
-                        if (!item.stats.isDirectory()) {
-                            var inputFile = item.path;
-                            var file = path.relative(inputDirectory, item.path);
-
-                            if (gzip && tilesOnly && !isTile(inputFile)) {
-                                copyFile(inputFile, file, writeFile);
-                            } else {
-                                isGzipped(inputFile)
-                                    .then(function(fileIsGzipped) {
-                                        if (fileIsGzipped === gzip) {
-                                            // File is already in the correct state
-                                            copyFile(inputFile, file, writeFile);
-                                        } else {
-                                            fsExtraReadFile(inputFile)
-                                                .then(function(data) {
-                                                    data = operation(data);
-                                                    writeFile(file, data);
-                                                })
-                                                .catch(reject);
-                                        }
-                                    })
-                                    .catch(reject);
-                            }
-                        }
-                    })
-                    .on('error', reject);
-            })
-            .catch(reject);
-    });
-}
-
-function isTile(file) {
-    var extension = path.extname(file);
-    return extension === '.b3dm' ||
-           extension === '.i3dm' ||
-           extension === '.pnts' ||
-           extension === '.cmpt' ||
-           extension === '.vctr';
-}
-
-function getNumberOfFilesInDirectory(directory) {
-    return new Promise(function(resolve, reject) {
-        var numberOfFiles = 0;
-        fsExtra.walk(directory)
-            .on('data', function (item) {
-                if (!item.stats.isDirectory()) {
-                    ++numberOfFiles;
+    return getFilesInDirectory(inputDirectory, {
```

Comment:
I used to do a similar approach here, but changed it for this comment: https://github.com/AnalyticalGraphicsInc/3d-tiles-tools/pull/3#r77263777. I'm fine with merging the changes in this file in though.

I have a branch that cleans up `gzipTileset` and the end result looks similar to this, but walks the directory without buffering the file names up front (through a helper function). And really that too is probably going to get blown away again at some later point.